pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -290,6 +309,31 @@ func (vm *windows) ConfigureKubeProxy(nodeName, hostSubnet string) error {
 
 // Interface helper methods
 
+// isHostNameChangeNeeded tells if we need to update the host name of the Windows VM
+func (vm *windows) isHostNameChangeNeeded() (bool, error) {
+	out, err := vm.Run("hostname", true)
+	if err != nil {
+		return false, errors.Wrapf(err, "error getting the host name for %s with stdout %s", vm.ID(), out)
+	}
+	return !strings.Contains(out, vm.hostName), nil
+}
+
+// changeHostName changes the hostName of the Windows VM to match with the machine name
+func (vm *windows) changeHostName() error {
+	changeHostNameCommand := "Rename-Computer -NewName " + vm.hostName + " -Force -Restart"
+	out, err := vm.Run(changeHostNameCommand, true)
+	if err != nil {
+		log.Info("changing host name failed", "command", changeHostNameCommand, "output", out,
+			"error", err)
+		return errors.Wrap(err, "changing host name failed")
+	}
+	// Since changing host name needs a restart, let's re-initialize the SSH connection
```

Comment:
`Reinitialize the SSH connection given changing the host name requires a VM restart`